### PR TITLE
modify build_electrodes function

### PR DESCRIPTION
**Pull request by JaerongA**
- modify build_electrodes function to make it usable for other types of probes.
- modify docstring for data types.

### CODE REVIEW

File: element_array_ephys/probe.py
Comment (dimitri-yatsenko) [inline element_array_ephys/probe.py:79]: hm.. does mysql even support non-ascii characters in comments?

File: element_array_ephys/probe.py
Comment (dimitri-yatsenko) [inline element_array_ephys/probe.py:259]:
```suggestion
    electrode_layouts = [
                {
                    "electrode": (site_count_per_shank * shank_no) + e_id,
                    "shank": shank_no,
                    "shank_col": c_id,
                    "shank_row": r_id,
                    "x_coord": x + (shank_no * shank_spacing),
                    "y_coord": y if y_origin == "bottom" else -y,
                }
                for shank_no in range(shank_count)
                for e_id, (c_id, r_id, x, y) in enumerate(
                    zip(shank_cols, shank_rows, x_coords, y_coords)
                )
            ]
```

File: element_array_ephys/probe.py
Comment (JaerongA) [reply element_array_ephys/probe.py:79]: I think so.. tables are created without any errors.

Comment (dimitri-yatsenko) [reply element_array_ephys/probe.py:79]: What do you get from `ProbeType.Electrode.describe()`?

**Comment by ttngu207** [comment]
Since we are revisiting this topic - I'd suggesting looking into using: https://github.com/SpikeInterface/probeinterface

**Comment by kabilar** [comment]
Thanks @JaerongA.  Are we holding off on merging, pending @ttngu207's suggestion above?  If so, let's draft this pull request.

Please update:
- [ ] version.py
- [ ] changelog.md

**Comment by JaerongA** [comment]
Ok, @ttngu207 and I haven't followed up on this. I'll convert this to draft in the meantime.

File: element_array_ephys/probe.py
Comment (ttngu207) [inline element_array_ephys/probe.py:152]:
```suggestion
                electrode_layouts, skip_duplicates=True
```

File: element_array_ephys/probe.py
Comment (ttngu207) [inline element_array_ephys/probe.py:255]:
```suggestion
            "y_coord": {"top": -y, "bottom": y}[y_origin],
```

**Review by ttngu207 — CHANGES_REQUESTED** [review]

**Review by ttngu207 — APPROVED** [review]

**Comment by JaerongA** [comment]
> Thanks @JaerongA. Are we holding off on merging, pending @ttngu207's suggestion above? If so, let's draft this pull request.
> 
> Please update:
> 
>     * [ ]  version.py
> 
>     * [ ]  changelog.md

Hi, @kabilar. I think this pr's good to go.  If you can review & merge this pending [pr](https://github.com/datajoint/element-array-ephys/pull/118), I'll update the version of this one accordingly. Thanks!

File: element_array_ephys/probe.py
Comment (CBroz1) [inline element_array_ephys/probe.py:224]:
```suggestion
        probe_type (str): probe type (e.g., "neuropixels 1.0 - 3A").
```

File: element_array_ephys/probe.py
Comment (CBroz1) [inline element_array_ephys/probe.py:239]: What motivated the use of `None` defaults here? Would it not have the same result with `1` as default and no `or` operators here?

If the following suggestion is accepted, please also change the `def` and docstring

```suggestion
        np.arange(0, col_spacing * col_count_per_shank, col_spacing),
        row_count,
    )
    y_coords = np.repeat(np.arange(row_count) * row_spacing, col_count_per_shank)
```

File: element_array_ephys/probe.py
Comment (CBroz1) [inline element_array_ephys/probe.py:257]: If using `1` defaults, this could also revert
```suggestion
            "x_coord": x + (shank_no * shank_spacing),
```

File: element_array_ephys/probe.py
Comment (CBroz1) [inline element_array_ephys/probe.py:250]: Without the old for-loop, you could bump the return up here and remove line 266
```suggestion
    return [
```

File: element_array_ephys/probe.py
Comment (CBroz1) [inline element_array_ephys/probe.py:150]: What's the benefit of adding `list[dict]` typing here? For functions, it might help someone understand the code later, but I don't think we're in the habit of doing so when grabbing results

**Review by CBroz1 — COMMENTED** [review]
Looks good overall. I had some questions about default choices and in-line typing suggestions. If these choices support the process in a way I don't understand, no need to incorporate my suggestions

File: element_array_ephys/probe.py
Comment (JaerongA) [reply element_array_ephys/probe.py:239]: While either scheme will yield the same results, I've purposefully set the default to `None` here because users may be misled to think there's an actual value associated with those (e.g., 1 mm spacing ?) when there isn't.

**Review by JaerongA — COMMENTED** [review]
Thanks @CBroz1 for the nice feedback. Here are my responses.

File: element_array_ephys/probe.py
Comment (CBroz1) [reply element_array_ephys/probe.py:239]: Makes sense. Thanks for your response

**Review by CBroz1 — COMMENTED** [review]
Thanks for your work on this, @JaerongA

File: element_array_ephys/probe.py
Comment (CBroz1) [inline element_array_ephys/probe.py:266]: Only remaining change would be to remove this unused return.
```suggestion
```

**Review by CBroz1 — APPROVED** [review]
Thanks for your work on this, @JaerongA! 

Please remember to push a tag to trigger a release

**Pull request merged**
